src/sagemaker/amazon/image_classification.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from sagemaker.amazon.amazon_estimator import AmazonS3AlgorithmEstimatorBase, registry
+from sagemaker.amazon.hyperparameter import Hyperparameter as hp  # noqa
+from sagemaker.amazon.validation import gt, isin, isint, ge, isstr, lt, le
+from sagemaker.predictor import RealTimePredictor
+from sagemaker.model import Model
+from sagemaker.session import Session
+
+
+class ImageClassification(AmazonS3AlgorithmEstimatorBase):
+
+    repo = 'image-classification:latest'
+
+    num_classes = hp('num_classes', (gt(1), isint), 'num_classes should be an integer greater-than 1')
+    num_training_samples = hp('num_training_samples', (gt(1), isint), 'num_training_samples should be an integer greater-than 1')
+    use_pretrained_model = hp('use_pretrained_model', (isin(0, 1), isint), 'use_pretrained_model should be in the set, [0,1]')
+    checkpoint_frequency = hp('checkpoint_frequency', (ge(1), isint), 'checkpoint_frequency should be an integer greater-than 1')
+    num_layers = hp('num_layers', (isin(18, 34, 50, 101, 152, 200, 20, 32, 44, 56, 110), isint), \
+                    'num_layers should be in the set [18, 34, 50, 101, 152, 200, 20, 32, 44, 56, 110]' )
+    resize = hp('resize', (gt(1), isint), 'resize should be an integer greater-than 1')
+    epochs = hp('epochs', (ge(1), isint), 'epochs should be an integer greater-than 1')
```

Comment:
Please note that isint() was removed, and the hp()'s have changed a bit. For reference see:
https://github.com/aws/sagemaker-python-sdk/pull/54